src/server/journal/journal.cc

```diff
@@ -60,10 +60,12 @@ error_code Journal::Close() {
 }
 
 uint32_t Journal::RegisterOnChange(ChangeCallback cb) {
+  lock_guard lk(state_mu_);
```

Comment:
Because now this flow preempts it might invalidate the iterators

Reply:
Also I will replace the `lock_guard` with `LocalBlockingCounter` but not in this PR. This is not that important for now

Reply:
But in this case we return the bug that we already fixed

Reply:
Why was this a bug?

Reply:
if we preempt during register on change we can lost data because db_slice and journal_slice should register on change without preemtion

Reply:
1) we did registerOnChange on db_slice
2) we call registerOnChange on journal and preempt
3) we add new key that affect db_slice and serialize bucket
4) we finish registerOnChange for journal and lost this key